src/pub.ts

```diff
@@ -0,0 +1,44 @@
+'use strict';
+
+import * as channels from "./channels";
+import * as child_process from "child_process";
+import * as os from "os";
+import * as path from "path";
+import * as project from "./project";
+import * as vs from "vscode";
+
+export class PubManager {
+	private sdk: string;
+
+	constructor(sdk: string) {
+		this.sdk = sdk;
+	}
+
+	registerCommands(context: vs.ExtensionContext) {
+		context.subscriptions.push(vs.commands.registerCommand("pub.get", selection => {
+			this.runPub("get", selection);
+		}));
+		context.subscriptions.push(vs.commands.registerCommand("pub.upgrade", selection => {
+			this.runPub("upgrade", selection);
+		}));
+	}
+
+	private runPub(command: String, selection?) {
+		let root = vs.workspace.rootPath;
+		let projectPath = selection
+			? path.dirname(selection.path)
+			: project.locateBestProjectRoot();
+		let shortPath = path.join(path.basename(root), path.relative(root, projectPath));
+		let channel = channels.getCreateChannel("Pub");
+		channel.show(true);
+
+		// TODO: Add a wrapper around the Dart SDK? It could do things like
+		// return the paths for tools in the bin/ dir. 
+		let pubPath = os.platform() == "win32"
```

Comment:
I have another platform check (for `dart.exe`/`dart`) so should probably consolidate these (I think the other is in `util`).

Actually, maybe this one isn't required (Windows checks certain extensions automatically, inc. `.bat`.. We don't need to use `dart.exe` when spwaning the server (only use it for symlink detection where we need the real filename).